packages/transport-dom/src/create.ts

```diff
@@ -164,17 +174,15 @@ export const createChannelTransport = ({
 
       if (method.kind === MethodKind.ServerStreaming) {
         const iter = input[Symbol.asyncIterator]();
-        const [{ value }, { done }] = [
-          (await iter.next()) as IteratorYieldResult<PartialMessage<I>>,
-          await iter.next(),
-        ];
-        if (!done)
+        const [{ value } = { value: null }, { done }] = [await iter.next(), await iter.next()];
+        if (done && typeof value === 'object' && value != null) {
+          const message = Any.pack(new method.I(value as object)).toJson(jsonOptions);
+          port.postMessage({ requestId, message, header } satisfies TransportMessage);
+        } else
```

Comment:
this removes the need for a very specific cast by using a default value, and then using a safer cast. the expected behavior is now confirmed with tests